scripts/remove_dev.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+userAgent="GitHub Autodeploy Bot/1.1.0 (${WIKI_UA_EMAIL})"
+
+. ./scripts/login_and_get_token.sh
+
+ckf="cookie_base.ck"
+declare -a removeErrors
+declare -a allWikis
+allWikis=$(
+  curl \
+    -s \
+    -b "$ckf" \
+    -c "$ckf" \
+    -H "User-Agent: ${userAgent}" \
+    -H 'Accept-Encoding: gzip' \
+    -X GET "https://liquipedia.net/api.php?action=listwikis" \
+    | gunzip \
+    | jq '.allwikis | keys[]' -r
+)
+# Don't get rate limited
+sleep 4
+
+removePage() {
+  page="${1}"
+  wiki=$2
+
+  echo "deleting ${wiki}:${page}"
+
+  wikiApiUrl="${WIKI_BASE_URL}/${wiki}/api.php"
+  ckf="cookie_${wiki}.ck"
+
+  getToken $wiki
+
+  rawRemoveResult=$(
+    curl \
+      -s \
+      -b "$ckf" \
+      -c "$ckf" \
+      --data-urlencode "title=${page}" \
+      --data-urlencode "reason=Remove ${LUA_DEV_ENV_NAME}" \
+      --data-urlencode "token=${token}" \
+      -H "User-Agent: ${userAgent}" \
+      -H 'Accept-Encoding: gzip' \
+      -X POST "${wikiApiUrl}?format=json&action=delete" \
+      | gunzip
+  )
+  # Don't get rate limited
+  sleep 8
+
+  if [[ $rawRemoveResult != *"delete"* ]]; then
+    echo "::warning::could not delete ${page} on ${wiki}"
+    echo "::warning::could not delete ${page} on ${wiki}" >> $GITHUB_STEP_SUMMARY
+    removeErrorMsg="${wiki}:${page}"
+    removeErrors+=("${removeErrorMsg}")
+  fi
+}
+
+searchAndRemove(){
+  wiki=$1
+
+  wikiApiUrl="${WIKI_BASE_URL}/${wiki}/api.php"
+
+  rawSearchResult=$(
+    curl \
+      -s \
+      -b "$ckf" \
+      -c "$ckf" \
+      --data-urlencode "list=search" \
+      --data-urlencode "srsearch=intitle:${LUA_DEV_ENV_NAME}" \
+      --data-urlencode "srnamespace=828" \
+      --data-urlencode "srlimit=500" \
```

Comment:
do we need to consider >500 results?
maybe (untested) via something like
```sh
if [[ ${#pages[@]} == 500 ]]; then
  if [[ $deletions == 0 ]]; then
    searchAndRemove $wiki 500
  else
    searchAndRemove $wiki 0
  fi
fi
```
(plus setting offset in the search accordingly and passing an additional 0 in line 96 plus adding a counter to check the number of deletions within a `searchAndRemove` call)

Reply:
You can just run the action twice if you really have that many pages

Reply:
what if you have a case where you only want to remove `hjp` but you have > 500 modules with `hjpX`?
(hence why i did suggested it with the offset)

Reply:
deploy is being done with a bot that has the `apihighlimits` right, so you should be able to query 5k things?